knora/dsplib/utils/onto_create_ontology.py

```diff
@@ -13,598 +13,618 @@
 
     Args:
         con: connection instance to connect to the DSP server
-        groups: List of definitions of the groups (JSON) to be created
-        project: Project the group(s) should be added to
-        verbose: Prints out more information if set to True
+        groups: "groups" section of a parsed JSON project file
+        project: Project the group(s) should be added to (must exist on DSP server)
 
     Returns:
-        Dict with group names and groups
+        dict of the form ``{group name: group object}`` with the groups that have successfully been created (or already exist). Empty dict if no group was created.
+        True if everything went smoothly, False if a warning or error occurred
     """
-    new_groups: dict[str, Group] = {}
+    overall_success = True
+    current_project_groups: dict[str, Group] = {}
+    try:
+        remote_groups: list[Group] = try_network_action(
+            object=Group,
+            method="getAllGroupsForProject",
+            kwargs={"con": con, "proj_shortcode": project.shortcode},
+            error_message_on_failure="WARNING: Unable to check if group names are already existing on DSP server, "
+                                     "because it is not possible to retrieve the remote groups from DSP server."
+        )
+    except BaseError as err:
+        print(err.message)
+        remote_groups = []
+        overall_success = False
+        
     for group in groups:
         group_name = group["name"]
 
-        # check if the group already exists, skip if so
-        all_groups: Optional[list[Group]] = Group.getAllGroups(con)
-
-        group_exists: bool = False
-        if all_groups:
-            group_exists = any(group_item.name == group_name for group_item in all_groups)
-
-        if group_exists:
-            print(f"WARN Group name '{group_name}' already in use. Skipping...")
+        # if the group already exists, add it to "current_project_groups" (for later usage), then skip it
+        remotely_existing_group = [g for g in remote_groups if g.name == group_name]
+        if remotely_existing_group:
+            current_project_groups[group_name] = remotely_existing_group[0]
+            print(f"\tWARNING: Group name '{group_name}' already exists on the DSP server. Skipping...")
+            overall_success = False
             continue
 
-        # check if status is defined, set default value if not
-        group_status: Optional[str] = group.get("status")
-        group_status_bool = True
-        if isinstance(group_status, str):
-            group_status_bool = json.loads(group_status.lower())  # lower() converts string to boolean
-
-        # check if selfjoin is defined, set default value if not
-        group_selfjoin: Optional[str] = group.get("selfjoin")
-        group_selfjoin_bool = False
-        if isinstance(group_selfjoin, str):
-            group_selfjoin_bool = json.loads(group_selfjoin.lower())  # lower() converts string to boolean
-
```

Comment:
I simplified these two radically: They are now directly in the constructor call below, as one-liners